From 82d2911f36fe1a97bc0e5dbe490ff2f12e742a5c Mon Sep 17 00:00:00 2001
From: Kaifeng Zhu <kaifeng.zhu@citrix.com>
Date: Fri, 7 Mar 2014 15:16:42 +0000
Subject: [PATCH 04/10] hw/device-hotplug: fix test of drive_add() return
To: andrew.cooper3@citrix.com

drive_opt_idx could be -1 in case error occurs inside drive_add, so the error
check should be "if (drive_opt_idx < 0)" instead of original
"if (!drive_opt_idx)".
Coverity issue id: 20279.

Signed-off-by: Kaifeng Zhu <kaifeng.zhu@citrix.com>
diff --git a/hw/device-hotplug.c b/hw/device-hotplug.c
index 3bdc048..97d3529 100644
--- a/hw/device-hotplug.c
+++ b/hw/device-hotplug.c
@@ -34,7 +34,7 @@ int add_init_drive(const char *opts)
     int ret = -1;
 
     drive_opt_idx = drive_add(NULL, "%s", opts);
-    if (!drive_opt_idx)
+    if (drive_opt_idx < 0)
         return ret;
 
     drive_idx = drive_init(&drives_opt[drive_opt_idx], 0, current_machine);
